Crash ``list index out of range`` on simple and wrong python

### Bug description

I tried to find this example but was unable to.  Please note that pylint some times crashes on this example that I collapsed from a larger example, as I have run it multiple times.  I am submitting this as pylint shouldn't crash on badly written python code and this is such a small example.



```python
""" Python example """

mydict={
    "one":"1",
    "two":"2",
    "three": "3"
}

for x, in mydict.items():
    {
    print(x,mydict[x])
    }

```

pylint crashed with a ``IndexError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1034, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1069, in _check_file
    check_astroid_module(ast_node)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1203, in check_astroid_module
    retval = self._check_astroid_module(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1250, in _check_astroid_module
    walker.walk(node)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 637, in visit_for
    self._check_unnecessary_dict_index_lookup(node)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1899, in _check_unnecessary_dict_index_lookup
    args=(node.target.elts[1].as_string()),
IndexError: list index out of range
```


### Configuration

_No response_

### Command used

```shell
pylint t001.py
```


### Pylint output

```shell
Exception on node <For l.9 at 0x7fbbb9ebd640> in file '/Users/Rob 1/Programming/python/openshift-acct-mgt/t001.py'
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 637, in visit_for
    self._check_unnecessary_dict_index_lookup(node)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1899, in _check_unnecessary_dict_index_lookup
    args=(node.target.elts[1].as_string()),
IndexError: list index out of range
************* Module t001
t001.py:1:0: F0001: Fatal error while checking 't001.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/Users/Rob 1/Library/Caches/pylint/pylint-crash-2021-12-10-20.txt'. (fatal)

------------------------------------
Your code has been rated at 10.00/10
```


### Expected behavior

I would expect something other than a "Fatal Error", perhaps something indicating that C programmers shouldn't be programming in Python :).

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.0 (v3.9.0:9cf6752276, Oct  5 2020, 11:29:23) 
[Clang 6.0 (clang-600.0.57)]
```


### OS / Environment

Mac OSX 11.6.1

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -25,6 +25,11 @@ Release date: TBA
 
   Closes #4716
 
+* Fix crash in ``unnecessary-dict-index-lookup`` checker if the output of
+  ``items()`` is assigned to a 1-tuple.
+
+  Closes #5504
+
 * The ``PyLinter`` class will now be initialiazed with a ``TextReporter``
   as its reporter if none is provided.
 

--- a/doc/whatsnew/2.13.rst
+++ b/doc/whatsnew/2.13.rst
@@ -26,6 +26,11 @@ Other Changes
 
   Closes #4716
 
+* Fix crash in ``unnecessary-dict-index-lookup`` checker if the output of
+  ``items()`` is assigned to a 1-tuple.
+
+  Closes #5504
+
 * Fix false negative for ``consider-iterating-dictionary`` during membership checks encapsulated in iterables
   or ``not in`` checks
 

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -1882,6 +1882,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                     if isinstance(value, nodes.Name):
                         if (
                             not isinstance(node.target, nodes.Tuple)
+                            # Ignore 1-tuples: for k, in d.items()
                             or len(node.target.elts) < 2
                             or value.name != node.target.elts[0].name
                             or iterating_object_name != subscript.value.as_string()

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -1882,6 +1882,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                     if isinstance(value, nodes.Name):
                         if (
                             not isinstance(node.target, nodes.Tuple)
+                            or len(node.target.elts) < 2
                             or value.name != node.target.elts[0].name
                             or iterating_object_name != subscript.value.as_string()
                         ):

--- a/tests/functional/u/unnecessary/unnecessary_dict_index_lookup.py
+++ b/tests/functional/u/unnecessary/unnecessary_dict_index_lookup.py
@@ -100,3 +100,10 @@ for key, val in outer_dict.items():
     for key_two, val_two in val.items():
         del outer_dict[key][key_two]  # [unnecessary-dict-index-lookup]
         break
+
+# Test partial unpacking of items
+# https://github.com/PyCQA/pylint/issues/5504
+
+d = {}
+for key, in d.items():
+    print(d[key])

